### PR TITLE
Update frontend to use status codes not IDs.

### PR DESCRIPTION
**Pull request by devinleighsmith**
Also correct bugs in frontend introduced by https://github.com/bcgov/PIMS/pull/333

### CODE REVIEW

File: backend/dal/Services/Concrete/ProjectService.cs [hunk 1]
```diff
@@ -521,6 +521,7 @@ public Project Update(Project project)
             var status = this.Context.ProjectStatus
                 .Include(s => s.Tasks)
                 .FirstOrDefault(s => s.Id == project.StatusId);
+            project.Status = status;
```
Comment (devinleighsmith) [inline backend/dal/Services/Concrete/ProjectService.cs:524]: Fixes an NPE thrown by project.IsProjectClosed(_options.Project)

File: frontend/src/components/common/form/Select.tsx [hunk 2]
```diff
@@ -121,7 +121,12 @@ export const Select: React.FC<SelectProps> = ({
         multiple={multiple}
         onChange={onSelectChange}
         onBlur={(e: any) => {
-          if (type === 'number' && !isNaN(parseInt(value))) {
+          if (type === 'number' && multiple) {
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/Select.tsx:124]: handle numeric multi-selects properly.

File: frontend/src/features/projects/dispose/hooks/useStepForm.tsx [hunk 3]
```diff
@@ -71,6 +71,7 @@ const useStepForm = () => {
       .catch((error: AxiosError) => {
         const msg: string = error?.response?.data?.error ?? error.toString();
         formikRef?.current?.setStatus({ msg });
+        throw Error('axios request failed');
```
Comment (devinleighsmith) [inline frontend/src/features/projects/dispose/hooks/useStepForm.tsx:74]: Ensures that add/update failures prevent navigation to the next step (hiding the API error).

File: backend/dal/Services/Concrete/ProjectService.cs (hunk 1 above)
Comment (Fosol) [reply backend/dal/Services/Concrete/ProjectService.cs:524]: Ah essentially manual lazy-load.

**Pull request merged**